python/mxnet/symbol.py

```diff
@@ -51,11 +51,31 @@ def __repr__(self):
                             'Grouped' if name is None else name)
 
     def __iter__(self):
-        """Returns all outputs in a list"""
+        """Returns a generator object of symbol.
+
+        One can loop through the returned object list to get outputs.
+
+        Example usage:
+        ----------
+        >>> a = mx.sym.Variable('a')
+        >>> b = mx.sym.Variable('b')
+        >>> c = a+b
+        >>> d = mx.sym.Variable('d')
+        >>> e = d+c
+        >>> out = e.get_children()
+        >>> out
+        <Symbol Grouped>
+        >>> for i in out.__iter__():
```

Comment:
No. you can choose not to call it explicitly. I called it to show the example call to the function. I will remove it